providers/default.rb

```diff
@@ -101,28 +101,29 @@ def mixlib_install
 # why would we use this when mixlib-install has a current_version method?
 # well mixlib-version parses the manifest JSON, which might not be there.
 # ohai handles this better IMO
+# @return String
 def current_version
   node['chef_packages']['chef']['version']
 end
 
-# the version we WANT TO INSTALL. If :latest is specified this will be the actual
-# latest version returned by mixlib-install. if download_url_override is passed
-# we parse the version, but blindly assume it's correct. Otherwise we rely on
-# mixlib-install to give us the exact version since someone could pass ~12 which
-# needs to be expanded to the latest 12.X.
+# the version we WANT TO INSTALL. If the user specifies a version in X.Y.X format
+# we use that without looking it up. If :latest or a non-X.Y.Z format version we
+# look it up with mixlib-install to determine the latest version matching the request
+# @return Mixlib::Versioning::Format::PartialSemVer
 def desired_version
-  if new_resource.download_url_override
-    # probably in an air-gapped environment.
-    version = Mixlib::Versioning.parse(new_resource.version)
-    Chef::Log.debug("download_url_override specified. Using specified version of #{version}")
```

Comment:
did you lose a use case here?

Reply:
Already fixed that one up. I think that was a bad rebase